app/contracts/work_packages/create_note_contract.rb

```diff
@@ -28,14 +30,16 @@
 
 module WorkPackages
   class CreateNoteContract < ::ModelContract
-    def self.model
-      WorkPackage
-    end
+    def self.model = WorkPackage
 
     attribute :journal_notes do
       errors.add(:journal_notes, :error_unauthorized) unless can?(:comment)
     end
 
+    attribute :journal_restricted do
+      errors.add(:journal_restricted, :invalid) if model.journal_notes.blank? && model.journal_restricted
+    end
```

Comment:
I think it makes things a bit simpler if we don't make a special case for notes. This check seems correct from the point of view of the "comments with restricted visibility" feature, but not really from journals.

Reply:
Keeping in mind this contract is a `CreateNoteContract` which is only used in the `AddWorkPackageNoteService` and not in any other journal creation service it would seem fitting. Pls shout if I'm missing some context

Reply:
I see. That makes much more sense now. :thinking:
All good, we can mark it as resolved :bow: